libpod/container_commit.go

```diff
@@ -0,0 +1,103 @@
+package libpod
+
+import (
+	"strings"
+
+	is "github.com/containers/image/storage"
+	"github.com/pkg/errors"
+	"github.com/projectatomic/libpod/libpod/buildah"
+	"github.com/projectatomic/libpod/libpod/image"
+	"github.com/sirupsen/logrus"
+)
+
+// ContainerCommitOptions is a struct used to commit a container to an image
+// It uses buildah's CommitOptions as a base. Long-term we might wish to
+// add these to the buildah struct once buildah is more integrated with
+//libpod
+type ContainerCommitOptions struct {
+	buildah.CommitOptions
+	Pause   bool
+	Author  string
+	Message string
+	Changes []string
+}
+
+// Commit commits the changes between a container and its image, creating a new
+// image
+func (c *Container) Commit(destImage string, options ContainerCommitOptions) (*image.Image, error) {
+	if !c.locked {
+		c.lock.Lock()
+		defer c.lock.Unlock()
+
+		if err := c.syncContainer(); err != nil {
+			return nil, err
+		}
+	}
+
+	if c.state.State == ContainerStateRunning && options.Pause {
+		if err := c.runtime.ociRuntime.pauseContainer(c); err != nil {
+			return nil, errors.Wrapf(err, "error pausing container %q", c.ID())
+		}
+		defer func() {
+			if err := c.runtime.ociRuntime.unpauseContainer(c); err != nil {
+				logrus.Errorf("error unpausing container %q: %v", c.ID(), err)
+			}
+		}()
+	}
+
+	sc := image.GetSystemContext(options.SignaturePolicyPath, "", false)
+	builderOptions := buildah.ImportOptions{
+		Container:           c.ID(),
+		SignaturePolicyPath: options.SignaturePolicyPath,
+	}
+	commitOptions := buildah.CommitOptions{
+		SignaturePolicyPath: options.SignaturePolicyPath,
+		ReportWriter:        options.ReportWriter,
+		SystemContext:       sc,
+	}
+	importBuilder, err := buildah.ImportBuilder(c.runtime.store, builderOptions)
```

Comment:
Does the container need to be mounted to do this?

Reply:
I dont know.  How can I tell?

Reply:
Does commit on a stopped container work? If it does, we should be fine

Reply:
yup it does